### PR TITLE
stats: speed up loading full stats info

### PR DESCRIPTION
**Pull request by lamxTyler**
Previously in #5215, it takes 1.5s to load full stats info when we have 800~ tables, however, it takes 1.2s to just execute the SQL statement for stats_buckets. Also, the `ExecRestrictedSQL` will first load all of the stats_buckets into memory, so it results in a lot more memory usage. In this PR, we use `Execute` and `Chunk` to reduce both the time and memory usage, it needs only 1s to load all stats info. However, if we do not use the `Chunk`, it needs about 1.3s to load all stats info.

### CODE REVIEW

**Comment by shenli** [comment]
Good job!

File: statistics/boostrap.go [hunk 1]
```diff
@@ -124,7 +150,7 @@ func (h *Handle) initStatsBuckets(tables statsCache) error {
 				continue
 			}
 			hist = &index.Histogram
-			lower, upper = row.GetDatum(6, &fields[6].Column.FieldType), row.GetDatum(7, &fields[7].Column.FieldType)
+			lower, upper = types.NewBytesDatum(row.GetBytes(6)), types.NewBytesDatum(row.GetBytes(7))
```
Comment (coocood) [inline statistics/boostrap.go]: The memory is a reference to the Chunk, and the Chunk will be reset and reused for next bytes.

**Comment by shenli** [comment]
@coocood PTAL

File: statistics/boostrap.go [hunk 2]
```diff
@@ -49,16 +46,31 @@ func (h *Handle) initStatsMeta(is infoschema.InfoSchema) (statsCache, error) {
...
 	}
-	return tables, nil
 }
 
-func (h *Handle) initStatsHistograms(is infoschema.InfoSchema, tables statsCache) error {
-	sql := "select table_id, is_index, hist_id, distinct_count, version, null_count, cm_sketch from mysql.stats_histograms"
-	rows, _, err := h.ctx.(sqlexec.RestrictedSQLExecutor).ExecRestrictedSQL(h.ctx, sql)
+func (h *Handle) initStatsMeta(is infoschema.InfoSchema) (statsCache, error) {
+	sql := "select version, table_id, modify_count, count from mysql.stats_meta"
+	rc, err := h.ctx.(sqlexec.SQLExecutor).Execute(goctx.TODO(), sql)
```
[the remark this answers is not in the document]
Comment (lamxTyler) [reply statistics/boostrap.go]: These two functions'  parameters are different.

**Comment by coocood** [comment]
LGTM

**Comment by shenli** [comment]
LGTM

**Comment by coocood** [comment]
@lamxTyler 
Please modify the test to make sure the test fails before the change to `newBytesDatum`.

**Comment by lamxTyler** [comment]
@coocood  Added.

**Pull request merged**